File: scenarios/codegen_exec/fixtures/fix_bug.py
from __future__ import annotations

import json
import sys
from pathlib import Path


def main() -> None:
    project_dir = Path(sys.argv[1])
    target = project_dir / "math_ops.py"
    source = target.read_text(encoding="utf-8")
    target.write_text(source.replace("return a - b", "return a + b"), encoding="utf-8")
    print(json.dumps({"patched_files": ["math_ops.py"], "workspace_dir": str(project_dir)}))


if __name__ == "__main__":
    main()

File: scenarios/codegen_exec/fixtures/fix_bug.ts
import { readFile, writeFile } from "node:fs/promises";
import path from "node:path";

const projectDir = process.argv[2];
if (!projectDir) {
  throw new Error("project directory is required");
}

const target = path.join(projectDir, "math_ops.ts");
const source = await readFile(target, "utf8");
await writeFile(target, source.replace("return a - b", "return a + b"), "utf8");

console.log(JSON.stringify({ patched_files: ["math_ops.ts"], workspace_dir: projectDir }));
